aurora/schemas/cycling.py

```diff
@@ -29,8 +29,8 @@ class CyclingParameter(GenericModel, Generic[DataT]):
     label: str  # the label used in a widget
     description: str = ""  # a long description
     units: str = ""  # physical units of this parameter
-    value: DataT = None  # the set value
-    default_value: DataT = None  # the default value of this parameter
+    value: DataT = None  # type: ignore[assignment] # the set value
+    default_value: DataT = None  # type: ignore[assignment] # the default value of this parameter
```

Comment:
Mmm just to make sure I understand correctly: is mypy complaining because you are assigning `None` to something that should be `DataT`? Why not use `Optional`?

Reply:
Excellent question. Optional is what one would do. However, when I did so, the app crashed :( This is because another part of the code (`technique_widget.py:build_parameter_widget`) checks the types of the `value` parameter. If it is not `str`, it raises a `NotImplementedYet` error. I tried understanding why the check is there but quickly lost patience, as the code is far too messy.

Reply:
I see 😂. Ok, interesting, could you open an issue documenting this? We need to figure out how to enable setting this to Optional or at least document somewhere why it is not possible.

Reply:
Thanks Loris. I figured as much. Just don't have the time for it at the moment :)

Reply:
As I understand it, the value of any `CyclingParameter` is optional, in which case the default value is assigned. However, can the default value be optional as well? That doesn't quite make sense to me.

Reply:
Okay. I found the issue. `Optional` makes the annotation of the attribute `Union[<actual-type>, NoneType]`. A simple solution then is to pick out the first part, i.e. the `<actual-type>`. The rest of the code then proceeds unfazed.

Comments?